sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -1339,6 +1340,31 @@ def test_issue_14787():
     f = Piecewise((x, x < 1), ((S(58) / 7), True))
     assert str(f.evalf()) == "Piecewise((x, x < 1), (8.28571428571429, True))"
 
+def test_issue_21481():
+    b, e = symbols('b e')
+    A = Piecewise((1, Eq(b, 1) | Eq(e, 0) | (Eq(b, -1) & Eq(Mod(e, 2), 0))),
+                    (0, Eq(b, 0) & (e > 0)), (-1, Eq(b, -1) & Eq(Mod(e, 2), 1)),
+                    (Piecewise((2, ((b > 1) & (e > 0)) | ((b > 0) & (b < 1) & (e < 0)) |
+                            ((e >= 2) & (b < -1) & Eq(Mod(e, 2), 0)) |
+                            ((e <= -2) & (b > -1) & (b < 0) & Eq(Mod(e, 2), 0))),
+                            (S.Half, ((b > 1) & (e < 0)) | ((b > 0) & (e > 0) & (b < 1)) |
+                            ((e <= -2) & (b < -1) & Eq(Mod(e, 2), 0)) |
+                            ((e >= 2) & (b > -1) & (b < 0) & Eq(Mod(e, 2), 0))),
+                            (-S.Half, Eq(Mod(e, 2), 1) & (((e <= -1) & (b < -1)) |
+                            ((e >= 1) & (b > -1) & (b < 0)))),
+                                (-2, ((e >= 1) & (b < -1) & Eq(Mod(e, 2), 1)) |
+                                ((e <= -1) & (b > -1) & (b < 0) & Eq(Mod(e, 2), 1)))),
+                    Eq(im(b), 0) & Eq(im(e), 0)))
```

Comment:
```suggestion
    C = Piecewise(
        (2,
         ((b > 1) & (e > 0)) |
         ((b > 0) & (b < 1) & (e < 0)) |
         ((e >= 2) & (b < -1) & Eq(Mod(e, 2), 0)) |
         ((e <= -2) & (b > -1) & (b < 0) & Eq(Mod(e, 2), 0))),
        (S.Half,
         ((b > 1) & (e < 0)) |
         ((b > 0) & (e > 0) & (b < 1)) |
         ((e <= -2) & (b < -1) & Eq(Mod(e, 2), 0)) |
         ((e >= 2) & (b > -1) & (b < 0) & Eq(Mod(e, 2), 0))),
        (-S.Half,
         Eq(Mod(e, 2), 1) &
         (((e <= -1) & (b < -1)) | ((e >= 1) & (b > -1) & (b < 0)))),
        (-2,
         ((e >= 1) & (b < -1) & Eq(Mod(e, 2), 1)) |
         ((e <= -1) & (b > -1) & (b < 0) & Eq(Mod(e, 2), 1)))
    )
    A = Piecewise(
        (1, Eq(b, 1) | Eq(e, 0) | (Eq(b, -1) & Eq(Mod(e, 2), 0))),
        (0, Eq(b, 0) & (e > 0)),
        (-1, Eq(b, -1) & Eq(Mod(e, 2), 1)),
        (C, Eq(im(b), 0) & Eq(im(e), 0))
    )
```
I just have this formatting suggestion